generator/standalone_main.cc

```diff
@@ -0,0 +1,132 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "google/cloud/internal/absl_str_join_quiet.h"
+#include "google/cloud/log.h"
+#include "google/cloud/status_or.h"
+#include "absl/flags/flag.h"
+#include "absl/flags/parse.h"
+#include "generator/generator.h"
+#include "generator/generator_config.pb.h"
+#include <google/protobuf/compiler/command_line_interface.h>
+#include <google/protobuf/text_format.h>
+#include <fstream>
+#include <future>
+#include <iostream>
+#include <sstream>
+
+ABSL_FLAG(std::string, config_file, "./generator/generator_config.textproto",
+          "Text proto configuration file specifying the code to be generated.");
+ABSL_FLAG(std::string, googleapis_commit_hash, "",
+          "Git commit hash of googleapis dependency.");
+ABSL_FLAG(std::string, protobuf_proto_path, "",
+          "Path to root dir of protos distributed with protobuf.");
+ABSL_FLAG(std::string, googleapis_proto_path, "",
+          "Path to root dir of protos distributed with googleapis.");
+ABSL_FLAG(std::string, output_path, ".",
+          "Path to root dir where code is emitted.");
+
+namespace {
+google::cloud::StatusOr<google::cloud::cpp::generator::GeneratorConfiguration>
+GetConfig(std::string const& filepath) {
+  std::ifstream input(filepath);
+  std::stringstream buffer;
+  buffer << input.rdbuf();
+
+  google::cloud::cpp::generator::GeneratorConfiguration generator_config;
+  auto parse_result = google::protobuf::TextFormat::ParseFromString(
```

Comment:
I think there's also a `ParseFromIstream()`, which may be simpler since you're starting with a `std::ifstream`